File: src/MyCounter.js
import { class_type } from "./fable_modules/fable-library.4.0.0-theta-018/Reflection.js";
import { Decorators_CustomElement_4E60E31B } from "./Aurelia.js";
import my_counter from "./my-counter.html";

export class MyCounter {
    constructor() {
        this.counter = 0;
    }
    increment() {
        const _ = this;
        _.counter = ((_.counter + 1) | 0);
    }
    decrement() {
        const _ = this;
        _.counter = ((_.counter - 1) | 0);
    }
}

export function MyCounter$reflection() {
    return class_type("MyCounter.MyCounter", void 0, MyCounter);
}

export function MyCounter_$ctor() {
    return new MyCounter();
}

Decorators_CustomElement_4E60E31B((() => {
    const template = my_counter;
    return {
        name: "my-counter",
        template: template,
    };
})())(MyCounter);

File: src/MyCounter.fs
module MyCounter

open Fable.Core

open Fable.Core.JsInterop
open type Aurelia.Decorators

[<AttachMembers>]
type MyCounter() =
    let mutable counter = 0

    member _.increment() : unit = counter <- counter + 1

    member _.decrement() : unit = counter <- counter - 1


CustomElement
    ({| template = importDefault "./my-counter.html"
        name = "my-counter" |})
    jsConstructor<MyCounter>
